### ISSUE TITLE
Using the saved model

### ISSUE THREAD

**Issue by Ethiral** [fchollet/keras#6865]
I've been training convolution autoencoder model inspired from the official documentation.I'm saving the intermediate layer which is encoding layer as a seperate model for future use.While loading the model I get a warning : No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '
I have saved the model as .h5 file with weights and architecture
Can I ignore the warning and if needed how do I compile the model manually?

Script:
#loading the input data

input_img = Input(shape=(28, 28,1))  

x = Conv2D(16, (3, 3), activation='relu', padding='same')(input_img)
x = MaxPooling2D((2, 2), padding='same')(x)
x = Conv2D(8, (3, 3), activation='relu', padding='same')(x)
x = MaxPooling2D((2, 2), padding='same')(x)
x = Conv2D(8, (3, 3), activation='relu', padding='same')(x)
encoded = MaxPooling2D((2, 2), padding='same')(x)

x = Conv2D(8, (3, 3), activation='relu', padding='same')(encoded)
x = UpSampling2D((2, 2))(x)
x = Conv2D(8, (3, 3), activation='relu', padding='same')(x)
x = UpSampling2D((2, 2))(x)
x = Conv2D(16, (3, 3), activation='relu')(x)
x = UpSampling2D((2, 2))(x)
decoded = Conv2D(1, (3, 3), activation='sigmoid', padding='same')(x)

autoencoder = Model(input_img, decoded)
autoencoder.compile(optimizer='adagrad', loss='binary_crossentropy')

encoder = Model(input_img, encoded)
encoder.save('encoder.h5')  

ThankYou

**Comment by kgrm** [comment]
It means you didn't run `encoder.compile()` before saving it, so it was saved without an optimizer. You can use it for inference, but if you want to train it further, you need to run the `compile` function after loading it.
